bin/i18n/sync-in.rb

```diff
@@ -98,6 +98,19 @@ def get_i18n_strings(level)
         name = behavior.at_xpath('./title[@name="NAME"]')
         i18n_strings['behavior_names'][name.content] = name.content if name
       end
+
+      text_blocks = blocks.xpath("//block[@type=\"text\"]")
+      i18n_strings['placeholder_texts'] = Hash.new unless text_blocks.empty?
+      text_blocks.each do |text_block|
+        text_title = text_block.at_xpath('./title[@name="TEXT"]')
+        # Skip empty or untranslatable string.
+        # A translatable string must have at least 3 consecutive alphabetic characters.
+        next unless text_title&.content =~ /[a-zA-Z]{3,}/
+
+        # Use only alphanumeric characters in lower cases as string key
+        text_key = text_title.content.gsub(/[^a-zA-Z0-9_ ]/, '').split.join('_').downcase
```

Comment:
I don't love the idea of inferring an identifier from the content of the string. When we've done things like this in the past, it ends up causing problems when the content changes and strings unexpectedly go missing, or when similar content is used in multiple places and the mapping ends up being non-unique.

Is there anything else we could use as a unique identifier here?

Reply:
How about using a MD5 hash? It will keep an 1:1 relationship between an ID and a string. 
Another option is to use a combination of script id, level id and string position, such as `script_11_level_399_str_1`. 
Did we use any of the above options in the past?

Reply:
We use string contents as IDs for `function_definitions` and `behavior_names`, is that because those strings are usually short and contain only alphabetic characters?
https://github.com/code-dot-org/code-dot-org/blob/49a8ed68830ce5bd1e262e14f12eae920cea926a/bin/i18n/sync-in.rb#L91
https://github.com/code-dot-org/code-dot-org/blob/49a8ed68830ce5bd1e262e14f12eae920cea926a/bin/i18n/sync-in.rb#L99

Reply:
It's because we were also unable to find a better option there. :upside_down_face:  Like I said, we've done this in the past but it's ended up being more fragile than we'd like.

An MD5 hash does address the issues of potential collisions, but we're still ending up with an identifier that's dependent on the content, rather than an identifier that can consistently identify content as it changes. That might be too much to ask for, though.

I'd love to see at least a mockup of the other end of this functionality; the code that's responsible for finding a translation given a block. I think that'll give us a better sense of which direction is best to go here.

Reply:
The sync-out and rendering pieces of this functionality is shorter than I thought so I add them to this PR. 

The rendering piece still uses string content as ID for now, just so we can verify it can render translations correctly.